src/webview/cluster/clusterViewLoader.ts

```diff
@@ -185,6 +190,22 @@ async function clusterEditorMessageListener (event: any ): Promise<any> {
     }
 }
 
+async function pollClipboard(signupStatus) {
+    while (panel) {
+        const previousContent = await vscode.env.clipboard.readText();
+        await new Promise(r => setTimeout(r, 500));
+        const currentContent = await vscode.env.clipboard.readText();
+        if (previousContent && previousContent !== currentContent) {
+            const oauthInfo = await sandboxAPI.getOauthServerInfo(signupStatus.apiEndpoint);
+            let errCode = '';
+            if (!await Cluster.validateLoginToken(signupStatus.apiEndpoint, currentContent)){
```

Comment:
Maybe validate login token can have the token check if it is SHA compatible or not.